content/300-guides/050-database/830-prismas-type-system.mdx

```diff
@@ -0,0 +1,152 @@
+---
+title: How to use Prisma's type system
+metaDescription: How to use Prisma's type system
+tocDepth: 2
+---
+
+<TopBlock>
+
+This guide introduces Prisma's type system and explains how to introspect existing native types in your database, and how to use types when you apply schema changes to your database with Prisma Migrate or `db push`.
+
+</TopBlock>
+
+## How does Prisma's type system work?
+
+Prisma uses _types_ to define the kind of data that a field can hold. To make it easy to get started, Prisma provides a small number of core [scalar types](/reference/api-reference/prisma-schema-reference#model-field-scalar-types) that should cover most default use cases. For example, in the blog post model below, the `title` field uses the `String` scalar type, while the `createdAt` field uses the `DateTime` scalar type:
+
+```prisma file=schema.prisma
+model Post {
+  id        Int      @id
+  title     String
+  createdAt DateTime
+}
+```
+
+Databases also have their own type system, which defines the type of value that a column can hold. Most databases provide a large number of data types to allow fine-grained control over exactly what a column can store. For example, a database might provide inbuilt support for multiple sizes of integers, or for XML data. The names of these types vary between databases. In PostgreSQL a variable length text column has a type of `text`, whereas in MySQL the type is `varchar`.
```

Comment:
PostgreSQL has a `varchar` type and MySQL has a `text` type accordingly. In Prisma, if the user does not define a native type for a `String` field, we use `text` on PostgreSQL and `varchar(...)` on MySQL.

The difference between `text` and `varchar` is that most of the time `text` is having an arbitrary length and is stored separately outside of the row structure, where `varchar` has a defined length and can be compressed for faster selection directly to the row. Of course SQL Server is a bit different here, and has additionally the `varchar(max)` and `nvarchar(max)` types that behave like the `text` type.

Reply:
Thanks - given this and Tom's comment above I've changed the example